### PR TITLE
Mccalluc/start qunit

### PR DESCRIPTION
**Pull request by mccalluc**
@thinkh or @sgratzl or @mstreit : I'd like to take a swing at some of the range issues (https://github.com/Caleydo/caleydo.github.io/issues/121 https://github.com/Caleydo/caleydo.github.io/issues/122 https://github.com/Caleydo/caleydo.github.io/issues/124
https://github.com/Caleydo/caleydo.github.io/issues/125 https://github.com/Caleydo/caleydo.github.io/issues/126 https://github.com/Caleydo/caleydo.github.io/issues/127
https://github.com/Caleydo/caleydo.github.io/issues/128 https://github.com/Caleydo/caleydo.github.io/issues/138) and I'd like to get a test framework in place first, so I can know that one fix isn't broken by the next.

The travis behavior right now is the same as what I see locally, and I think there are two problems.
- Am I compiling the typescript correctly? Should there be fewer errors?
- and relatedly: How do I use the compiled JS on a page? Is require necessary?

Feel free to add to this branch, if that's easier than explaining.

(Also, if you have opinions about test frameworks, let me know? This is mostly the same as what I'm doing for the tutorial tests on caleydo.github.io, but there are other options: I spent most of today looking at intern, and never quite got it to work, but I've pushed the branch with my work.)

### CODE REVIEW

**Comment by mccalluc** [comment]
@thinkh / @sgratzl : Got the tsc build for individual files working! The biggest obstacle to a merge is probably main.ts:

``` diff
-/// <reference path="../../tsd.d.ts" />
-/// <amd-dependency path="module" name="module"/>
+/// <reference path="typings/globals/es6-shim/index.d.ts"/>
```

The `../../` will not work if this checkout is the top level, instead of assuming there's a container. I don't understand the larger build process well enough to know how big a problem this will be.

I'm going to extend the tests and then, hopefully, start fixing some bugs: So I'm not blocked, and I'd be happy to merge even this much in, but I do need your help figuring out how this changes the larger build.

File: stratification.ts [hunk 1]
```diff
@@ -135,7 +135,7 @@ export class StratificationGroup extends idtypes.SelectAble implements IStratifi
     });
   }
 
-  idView(idRange:ranges.Range = ranges.all()):Promise<IStratification> {
+  idView(idRange:ranges.Range = ranges.all()):Promise<any> {
```
Comment (mccalluc) [inline stratification.ts:138]: Not sure if there's a better solution: 
`stratification.ts(139,12): error TS2322: Type 'Promise<void>' is not assignable to type 'Promise<IStratification>'.`

File: range.ts [hunk 2]
```diff
@@ -1242,9 +1242,13 @@ function parseRange(code:string) {
         dims.push(Range1D.all());
         break;
       default:
-        t = parseRange1D(code, act);
-        act = t.act + 1; //skip ,
-        dims.push(t.dim);
+        if (c.match(/\s/)) {
+          act++;
```
Comment (mccalluc) [inline range.ts:1246]: This is the only actual bug fix in this whole PR.

**Comment by mccalluc** [comment]
@thinkh or @sgratzl or @mstreit: I could use feedback on this; Does it look workable? If a call would be helpful, let me know. Will shift gears and then come back to this.

**Comment by mccalluc** [comment]
My focus right now is stubbing out the tests so we can at least get a sense of what good test coverage would imply. I wouldn't be surprised, and would actually hope, that the structure will change, perhaps to match the namespacing of web_bundle, but I'm pretty confident that the tests inside will still be useful.

Push tests pass, but [PR tests fail](https://travis-ci.org/Caleydo/caleydo_core/builds/165647088#L349) right now:

```
Died on test #1 exports.test/<@http://localhost:5000/tests/datatype_test.js:18:7: v is undefined
```

Will merge from master and see if that helps...

**Comment by mccalluc** [comment]
```
$ find . | grep .todo | wc -l
       7
$ grep TODO -r . | wc -l
     230
```

There are plenty of methods whose intended behavior I can understand, and so I'll start with them.

@sgratzl / @thinkh : 
- Perhaps the most useful thing you could do here would be to point out any methods that are either vestigial, or that ought to be private, so that I don't waste time writing tests for them. Just change the corresponding TODO to say that the method needs to be removed. (If nothing comes to mind, great.)
- Second priority is https://github.com/Caleydo/bundle_lib/issues/1#issuecomment-252056126

@ngehlenborg : There's plenty of low level work I can do here, and by re-grepping we can get a sense of the rate of progress. We can check-in next week and see if this approach makes sense.

Heading out early this afternoon, and then working for at least a little bit Monday.

File: tests/tests/geom_test.js [hunk 3]
```diff
@@ -73,16 +73,26 @@ define(["require", "exports", 'geom'], function (require, exports, geom) {
...
+            }
+          }
+          assert_intersect(1,1,1); // same
+          assert_intersect(-1,-1,1, false); // outside
+          assert_intersect(-1,1,1); // touch
+          assert_intersect(0,0,1); // overlap
+          // TODO: Bug? Maybe intersection means the line of the circle edge,
+          // rather than the interior of the disk?
+          assert_intersect(1,1,0.5, false); // smaller
+          assert_intersect(1,1,2, false); // larger
```
Comment (mccalluc) [inline tests/tests/geom_test.js:95]: @thinkh or @sgratzl : Does it look like we have the correct behavior here? For a sense of the bigger picture, can you point me at an example that uses geom, and in particular intersection?

**Comment by mccalluc** [comment]
@sgratzl : Before trying to get it working on Travis, I would like to get it working locally. Here are the errors I see right now:

```
caleydo_core$ npm list
caleydo_core@0.0.1-alpha /Users/chuck/github/caleydo/caleydo_core
├─┬ tslint@3.15.1
│ ├── colors@1.1.2
│ ├── diff@2.2.3
│ ├─┬ findup-sync@0.3.0
│ │ └── glob@5.0.15
│ ├─┬ glob@7.1.1
│ │ ├── fs.realpath@1.0.0
│ │ ├─┬ inflight@1.0.6
│ │ │ └── wrappy@1.0.2
│ │ ├── inherits@2.0.3
│ │ ├─┬ minimatch@3.0.3
│ │ │ └─┬ brace-expansion@1.1.6
│ │ │   ├── balanced-match@0.4.2
│ │ │   └── concat-map@0.0.1
│ │ ├── once@1.4.0
│ │ └── path-is-absolute@1.0.1
│ ├─┬ optimist@0.6.1
│ │ ├── minimist@0.0.10
│ │ └── wordwrap@0.0.3
│ ├── resolve@1.1.7
│ └─┬ underscore.string@3.3.4
│   ├── sprintf-js@1.0.3
│   └── util-deprecate@1.0.2
└── typescript@2.0.3

caleydo_core$ npm run compile

> caleydo_core@0.0.1-alpha compile /Users/chuck/github/caleydo/caleydo_core
> tsc

node_modules/typescript/lib/lib.es2015.core.d.ts(17,14): error TS2300: Duplicate identifier 'PropertyKey'.
typings/globals/es6-shim/index.d.ts(3,14): error TS2300: Duplicate identifier 'PropertyKey'.
typings/globals/es6-shim/index.d.ts(7,5): error TS2687: All declarations of 'value' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(245,5): error TS2687: All declarations of 'EPSILON' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(280,5): error TS2687: All declarations of 'MAX_SAFE_INTEGER' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(287,5): error TS2687: All declarations of 'MIN_SAFE_INTEGER' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(343,5): error TS2687: All declarations of 'flags' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(495,5): error TS2687: All declarations of 'prototype' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(558,5): error TS2687: All declarations of 'size' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(567,5): error TS2687: All declarations of 'prototype' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(578,5): error TS2687: All declarations of 'size' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(588,5): error TS2687: All declarations of 'prototype' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(603,5): error TS2687: All declarations of 'prototype' must have identical modifiers.
typings/globals/es6-shim/index.d.ts(617,5): error TS2687: All declarations of 'prototype' must have identical modifiers.

npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.8.0/bin/node" "/usr/local/bin/npm" "run" "compile"
npm ERR! node v6.8.0
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! caleydo_core@0.0.1-alpha compile: `tsc`
npm ERR! Exit status 2
```

My versions do not match what is on travis:
https://travis-ci.org/Caleydo/caleydo_core/builds/167392668#L201

```
$ node --version
v0.10.36
0.24s$ npm --version
1.4.28
0.14s$ tsc --version
Version 2.0.3
```

... but at least for node, that version is not officially supported, so I'm not sure if that's the best thing to upgrade.

Mostly just wanted to ask if the errors suggest a particular component to you, but I don't feel really blocked, yet.

**Comment by sgratzl** [comment]
there is no 'typings' directory anymore nor 'typings.json' in the repository, just delete the local directory

**Comment by mccalluc** [comment]
```
gems/selenium-webdriver-3.0.0/lib/selenium/webdriver/firefox.rb:58:in `driver_path':         Unable to find Mozilla geckodriver
```

This doesn't seem to be related to any of our changes. Weird. Will restart.

**Comment by mccalluc** [comment]
Closing this PR in favor of #6, which gets tests passing on saucelabs.

**Pull request closed without merging**